test/check_output.py

```diff
@@ -1,9 +1,16 @@
-files = ["first_single_beat.log", "second_single_beat.log", "third_single_beat.log"]
+files = [
+    "first_single_beat.log",
+    "second_single_beat.log",
+    "third_single_beat.log"
+]
 
 hello_cnt = 0
 for fname in files:
-    f = open(fname)
-    if "hello - " in f.read():
-        hello_cnt += 1
+    try:
+        f = open(fname)
+        if "hello - " in f.read():
+            hello_cnt += 1
+    except OSError:
```

Comment:
As the subprocess in `test-single-beat.sh` now handles the file redirection, the output for the processes which are not the leader will not create log files.